### DISCUSSION TITLE
AUTH AZURE AD returns AADSTS9002325: Proof Key for Code Exchange is required for cross-origin authorization code redemption.

### DISCUSSION THREAD

**Discussion by spacepirate0001** [langfuse/langfuse#2415] [Support] [answered]
I am running LangFuse as self-hosted in Azure and I already set application authentication with callback url:https://*.*.azurecontainerapps.io/api/auth/callback/azure-ad. When I try to sign-in I get the following error: `AADSTS9002325: Proof Key for Code Exchange is required for cross-origin authorization code redemption.` What am I doing wrong? I already followed the docs https://langfuse.com/docs/deployment/self-host#sso and added

AUTH_AZURE_AD_CLIENT_ID
AUTH_AZURE_AD_CLIENT_SECRET
AUTH_AZURE_AD_TENANT_ID

**Comment by marcklingen** [comment, marked as answer]
based on [this thread](https://learn.microsoft.com/en-us/answers/questions/1194524/how-to-fix-aadsts9002325-proof-key-for-code-exchan), this could be the answer:

> The redirect_uri of the external sso / idp should be registered as a "Web" instead of SPA.
